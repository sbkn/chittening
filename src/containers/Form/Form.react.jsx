import React, {Component} from "react";
import {Field, reduxForm} from "redux-form/immutable";
import Button from "../../components/Button/Button.react.jsx";

const renderInput = ({input, type, label}) => (
	<div>
		<label>{label}</label>
		<input {...input} type={type}/>
	</div>
);

class Form extends Component {

	render() {
		return (
			<form onSubmit={() => this.props.handleSubmit(() => {})}>

				<Field name="fieldUsername"
				       type="text"
				       label="username"
				       component={renderInput}
				/>

				<Field name="fieldPassword"
				       type="password"
				       label="password"
				       component={renderInput}
				/>

				<Button type="button" onClick={() => this.props.handleSubmit(() => {})}/>

			</form>
		)
	}
}


export default reduxForm()(Form);
