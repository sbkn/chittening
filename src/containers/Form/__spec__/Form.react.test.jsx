import React from "react";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import Form from "../Form.react.jsx";

describe("Form", () => {

	it("should render correctly", () => {

		const mockStore = {
			subscribe: () => {},
			dispatch: () => {},
			getState: () => {}
		};

		const tree = renderer.create(
			<Provider store={ mockStore }>
				<Form form="mockFormName"/>
			</Provider>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});