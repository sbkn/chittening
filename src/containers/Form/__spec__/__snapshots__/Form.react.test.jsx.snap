// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label>
      username
    </label>
    <input
      name="fieldUsername"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label>
      password
    </label>
    <input
      name="fieldPassword"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</form>
`;
